src/client.h

```diff
@@ -445,12 +445,25 @@ class Client : public con::PeerHandler, public InventoryManager, public IGameDef
 	Inventory* getInventory(const InventoryLocation &loc);
 	void inventoryAction(InventoryAction *a);
 
-	// Gets closest object pointed by the shootline
-	// Returns NULL if not found
+	/*!
+	 * Gets closest object pointed by the shootline.
+	 * Returns NULL if not found.
+	 * @param[in]  shootline_on_map    the shootline for
+	 * the test in world coordinates
+	 *
+	 * @param[out] intersection_point  the first point where
+	 * the shootline meets the object. Valid only if
+	 * not NULL is returned.
+	 *
+	 * @param[out] intersection_normal the normal vector of
+	 * the intersection, pointing outwards. Zero vector if
+	 * the shootline starts in an active object.
+	 * Valid only if not NULL is returned.
+	 */
 	ClientActiveObject * getSelectedActiveObject(
-			f32 max_d,
-			v3f from_pos_f_on_map,
-			core::line3d<f32> shootline_on_map
+		const core::line3d<f32> &shootline_on_map,
+		v3f &intersection_point,
```

Comment:
can you use pointers instead of non const references? non const references are forbidden by the code style.